src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -225,9 +224,9 @@ public static void UnscheduleFromRunLoop(SafeEventStreamHandle eventStream)
                     }
                 }
 
-                private static void WatchForFileSystemEventsThreadStart(object args)
+                private static void WatchForFileSystemEventsThreadStart(object? args)
                 {
-                    var inputArgs = (object[])args;
+                    var inputArgs = (object[])args!;
```

Comment:
NIT isn't it better bang on the caller

Reply:
A `!` doesn't work at the call site (it's passing a method group as a delegate and would need to use a pragma instead), but I can restructure things to make this a little cleaner.